services/order.py

```diff
@@ -0,0 +1,35 @@
+from django.db import transaction
+from django.db.models import QuerySet
+from django.contrib.auth import get_user_model
+
+from db.models import Order, Ticket
+
+
+User = get_user_model()
+
+
+def create_order(tickets: list[dict], username: str, date: str = None) -> None:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+
+        order = Order.objects.create(user=user)
+
+        if date:
+            order.created_at = date
+
+        order.save()

```

Comment:
This piece of code seems to violate the 'Don't Repeat Yourself' principle from the checklist. You are calling `order.save()` twice - once in line 15 and once in line 20. It would be better to set all the attributes of the `order` object first, and then call `save()` just once.